### PR TITLE
Add bindings and functions to work with linum margin.

### PR DESCRIPTION
**Pull request by ralesi**
- single click selects line
- double click selects text block
- drag across lines selects all lines dragged across

sorry for all the PR ;)  I have been sitting on them for a month.

### CODE REVIEW

File: spacemacs/keybindings.el [hunk 1]
```diff
@@ -413,3 +413,10 @@ Ensure that helm is required before calling FUNC."
   ("w" other-window                          :doc (spacemacs//window-manipulation-move-doc)))
 
 ;; end of Window Manipulation Micro State
+
+;; linum margin bindings ---------------------------------------------------------------
+
+(global-set-key (kbd "<left-margin> <down-mouse-1>") 'spacemacs/md-select-linum)
+(global-set-key (kbd "<left-margin> <mouse-1>") 'spacemacs/mu-select-linum)
+(global-set-key (kbd "<left-margin> <double-mouse-1>") 'spacemacs/select-current-block)
+(global-set-key (kbd "<left-margin> <drag-mouse-1>") 'spacemacs/mu-select-linum)
```
Comment (syl20bnr) [inline spacemacs/keybindings.el]: Move this block of global bindings at the top of the file before the leader section.

File: spacemacs/funcs.el [hunk 2]
```diff
@@ -143,6 +143,55 @@ the current state and point position."
...
+
+(defun spacemacs/md-select-linum (event)
+  (interactive "e")
+  (mouse-select-window event)
+  (goto-line (spacemacs/line-at-click))
+  (set-mark (point))
+  (setq *linum-mdown-line*
+        (line-number-at-pos)))
+
+(defun spacemacs/mu-select-linum ()
```
Comment (syl20bnr) [inline spacemacs/funcs.el]: Missing documentation

File: spacemacs/funcs.el [hunk 3]
```diff
@@ -143,6 +143,55 @@ the current state and point position."
...
+  (save-excursion
+    (let ((click-y (cddr (mouse-position)))
+          (debug-on-error t)
+          (line-move-visual t))
+      (goto-char (window-start))
+      (next-line (1- click-y))
+      (1+ (line-number-at-pos))
+      )))
+
+(defun spacemacs/md-select-linum (event)
```
Comment (syl20bnr) [inline spacemacs/funcs.el]: Missing documentation

File: spacemacs/funcs.el [hunk 4]
```diff
@@ -143,6 +143,55 @@ the current state and point position."
       (sp-newline)
       (setq counter (1- counter)))))
 
+
+;; linum gutter helpers
+(defvar *linum-mdown-line* nil)
+
+(defun spacemacs/line-at-click ()
```
Comment (syl20bnr) [inline spacemacs/funcs.el]: Missing documentation

**Comment by ralesi** [comment]
Updated with better documentation.

**Comment by syl20bnr** [comment]
Very cool addition :+1: 
Thank you !
Cherry-picked into develop branch, you can safely delete your branch.

**Pull request closed without merging**